preloader.js

```diff
@@ -0,0 +1,103 @@
+'use strict'
+const util = require('util')
+const toFastProperties = require('to-fast-properties')
+const filter = new Set(Object.keys(require.cache))
```

Comment:
I don't think this object is needed outside of initialization,  would you mind wrapping its use in a function so it can be garbage collected?